From b78a19e114a251cd25c501373efaa8ed4574259f Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Thu, 16 Oct 2014 16:02:14 -0400
Subject: [scsi] mpt2sas: Use pci_enable_msix_exact() instead of pci_enable_msix()

Message-id: <1413475335-14737-32-git-send-email-thenzl@redhat.com>
Patchwork-id: 97607
O-Subject: [RHEL7.1 e-stor PATCH 31/32] mpt2sas: Use pci_enable_msix_exact() instead of pci_enable_msix()
Bugzilla: 1088524
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range()  or pci_enable_msi_exact()
and pci_enable_msix_range() or pci_enable_msix_exact()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Reviewed-by: Tomas Henzl <thenzl@redhat.com>
Cc: Nagalakshmi Nandigama <Nagalakshmi.Nandigama@lsi.com>
Cc: Sreekanth Reddy <Sreekanth.Reddy@lsi.com>
Cc: support@lsi.com
Cc: DL-MPTFusionLinux@lsi.com
Cc: linux-scsi@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 52674c65f9751f607a4ed9d75227a0d8e4f54189)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/mpt2sas/mpt2sas_base.c b/drivers/scsi/mpt2sas/mpt2sas_base.c
index 32cd7fa..58e4521 100644
--- a/drivers/scsi/mpt2sas/mpt2sas_base.c
+++ b/drivers/scsi/mpt2sas/mpt2sas_base.c
@@ -1454,10 +1454,10 @@ _base_enable_msix(struct MPT2SAS_ADAPTER *ioc)
  for (i = 0, a = entries; i < ioc->reply_queue_count; i++, a++)
   a->entry = i;
 
- r = pci_enable_msix(ioc->pdev, entries, ioc->reply_queue_count);
+ r = pci_enable_msix_exact(ioc->pdev, entries, ioc->reply_queue_count);
  if (r) {
-  dfailprintk(ioc, printk(MPT2SAS_INFO_FMT "pci_enable_msix "
-      "failed (r=%d) !!!\n", ioc->name, r));
+  dfailprintk(ioc, printk(MPT2SAS_INFO_FMT
+      "pci_enable_msix_exact failed (r=%d) !!!\n", ioc->name, r));
   kfree(entries);
   goto try_ioapic;
  }
-- 
1.7.1